compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1315,7 +1315,11 @@ fn should_encode_fn_sig(def_kind: DefKind) -> bool {
 
 fn should_encode_constness(def_kind: DefKind) -> bool {
     match def_kind {
-        DefKind::Fn | DefKind::AssocFn | DefKind::Closure | DefKind::Ctor(_, CtorKind::Fn) => true,
+        DefKind::Fn
+        | DefKind::AssocFn
+        | DefKind::Closure
+        | DefKind::Ctor(_, CtorKind::Fn)
+        | DefKind::Impl { of_trait: false } => true,
```

Comment:
I see something asymmetric here, and this is something we might want to address either here or some time later.

We only encode impl methods' constness for trait impls and those are inherited from the impl, but is it not better to just look at the parent and only encode constness once? Since we're going with the whole-trait whole-impl approach anyways.

In the other direction, we should not encode `const impl` also and have all the assocs encode their constness.

Reply:
I will look into this metadata encoding thing later